src/data-access/LangCodeRetriever.ts

```diff
@@ -0,0 +1,10 @@
+export default interface LangCodeRetriever {
+
+	/*
+	 * Returns the language code associated with the Item,
+	 * which may or may not be a valid language code, validation needs to be done by the caller
+	 * returns `null` if the Item does not have a language code associated with it
+	 * returns `false` if the Item has novalue or somevalue for the language code property
```

Comment:
I think there should be some punctuation here, otherwise it’s going to look weird in some IDEs (“…associated with it returns `false` if…”).
```suggestion
	 * Returns the language code associated with the Item,
	 * which may or may not be a valid language code, validation needs to be done by the caller.
	 * Returns `null` if the Item does not have a language code associated with it.
	 * Returns `false` if the Item has novalue or somevalue for the language code property.
```